### PR TITLE
Fix for Optional model validation during parameter validation

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Context on - https://github.com/kedro-org/kedro/pull/5542

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: kedro/validation/parameter_validator.py [hunk 1]
```diff
@@ -50,16 +56,23 @@ def _apply_validation(self, raw_params: dict, requirements: dict) -> dict:
             try:
                 raw_value = resolve_nested_dict_path(raw_params, param_key)
 
-                if raw_value is None:
+                if raw_value is _MISSING:
```
Comment (PragnyaKhandelwal) [inline kedro/validation/parameter_validator.py:59]: The `_MISSING` / `None` split looks right, and the optional handling is much clearer now. I’d still like one end-to-end regression test through `validate_raw_params()` for `Optional[SamplePydanticModel] = None`, so we cover the full extraction + validation path rather than only `_apply_validation()`.

File: tests/validation/test_parameter_validator.py [hunk 2]
```diff
@@ -129,3 +129,17 @@ def test_does_not_modify_original(self, parameter_validator):
 
         parameter_validator._apply_validation(raw, requirements)
         assert isinstance(raw["config"], dict)
+
```
Comment (PragnyaKhandelwal) [inline tests/validation/test_parameter_validator.py:132]: The new unit tests cover the edge case well, but they only exercise `_apply_validation()` directly. Here we can add one integration-style test that builds a small pipeline node with `Optional[SamplePydanticModel]` and asserts `validate_raw_params({"model_options": None})` returns `None` instead of raising.

**Review by PragnyaKhandelwal — CHANGES_REQUESTED** [review]

**Comment by ravi-kumar-pilla** [comment]
Hi @PragnyaKhandelwal ,

This is a draft PR for your reference. I am happy to take this forward, but if you would like to introduce these changes in your PR [here](https://github.com/kedro-org/kedro/pull/5542) it would be great. Thank you

**Comment by ravi-kumar-pilla** [comment]
Closing as this is only a reference impl
